nens_auth_client/backends.py

```diff
@@ -56,11 +60,20 @@ def authenticate(self, request, claims):
         Returns:
           user or None
         """
+        allow_username_match = int(claims.get("custom:from_sso", 0)) == 1
         username = claims.get("cognito:username")
-        if not username:
-            return
-        allow_username_match = claims.get("custom:from_sso", 0)
-        if int(allow_username_match) != 1:  # AWS formats integers as strings
+        if not allow_username_match:
+            try:
+                assert claims["identities"][0]["providerName"] == "Google"
+                assert claims["email_verified"]
+                username, domain = claims["email"].split("@")
+                assert domain == "nelen-schuurmans.nl"
+            except (KeyError, IndexError, AssertionError, ValueError):
```

Comment:
Asserts zijn voor dingen die altijd waar zijn, niet om een soort if/else te doen.
Even snel `claims["identities"][0]["providerName"]` doen en dat met keyerror/indexerror afvangen: ok, prima.

Ik zou er een `if is_nens_from_google(claims):` functie van maken, dan kan je makkelijk drie keer `if not claims....: return` doen i.p.v. die asserts.

Reply:
Ik heb het in een functie gezet, zie laatste commit